modules/gatsby/src/components/layout/header.jsx

```diff
@@ -53,36 +52,106 @@ import {Link} from 'react-router-dom';
 })
 */
 
+function hasExamples(props) {
+  const { config = {} } = props;
+  const { EXAMPLES } = config;
+  return !(EXAMPLES.length === 0 || EXAMPLES[0].title === 'none');
+}
+
+function HeaderLink({ to, href, label, classnames }) {
+  if (to) {
+    return (
+      <Link to={to} className={classnames}>
+        {label}
+      </Link>
+    );
+  }
+  return (
+    <a href={href} className={classnames}>
+      {label}
+    </a>
+  );
+}
+
 export default class Header extends Component {
+  constructor(props) {
+    super(props);
+    // we need to know the number of links before render.
+    // this is not an ideal solution.
+    // some of the links which are hardcoded should come from configuration
+
+    {
+      /*
+            <Link className={classNames({active: pathname === '/search'})} to="/search">Search</Link>
+            Object.keys(trees).map(p => (
+              <Link
+                className={classNames({active: pathname.includes(p)})}
+                to={p}
+                key={p}
+              >
+                {trees[p].name}
+              </Link>
+            )) */
+    }
 
-  componentWillReceiveProps(nextProps) {
-    if (this.props.pathname === nextProps.pathname) {
-      return;
+    {
+      /* ADDITIONAL_LINKS.map(link => (
+              <a key={link.href} href={link.href}>{link.name}</a>
+            )) */
     }
-    // this.props.setHeaderOpacity(1);
-    // this.props.toggleMenu(false);
+
+    this.state = {
+      links: [
+        ...(hasExamples(props) ? [{ label: 'Examples', to: '/examples' }] : []),
+        { label: 'Documentation', to: '/docs' },
+        { label: 'Blog', href: 'https://medium.com/@vis.gl' },
+        ...(props.config && props.config.PROJECT_TYPE === 'github'
+          ? [
+              {
+                classnames: 'z',
+                href: props.config.PROJECT_URL,
+                label: (
+                  <div className="github-link">
+                    <span>Github</span>
+                    <GithubIcon
+                      style={{ marginLeft: '0.5rem', display: 'inline' }}
+                    />
+                    {/* <span className="Stars fac fje">
+                      {props.githubLoading ? '...' : props.stargazers_count}
+                      <StarIcon style={{ marginLeft: '0.5rem', display: 'inline' }} />
+                    </span> */}
+                  </div>
+                )
+              }
+            ]
+          : [])
+      ]
+    };
   }
 
   renderHeader() {
     // TODO/ib - replace data with config
-    const {config = {}, pathname, isMenuOpen, opacity, stargazers_count, githubLoading} = this.props;
+    const {
```

Comment:
You can remove the TODO

Reply:
I'm going to make one more pass to include github stars